saleor/graphql/checkout/mutations.py

```diff
@@ -458,15 +478,21 @@ class Meta:
 
     @classmethod
     def validate_checkout_lines(
-        cls, variants, quantities, country, channel_slug, lines=None
+        cls, request, variants, quantities, country, channel_slug, lines=None
     ):
         check_lines_quantity(
-            variants, quantities, country, channel_slug, existing_lines=lines
+            variants,
+            quantities,
+            country,
+            channel_slug,
+            existing_lines=lines,
+            check_reservations=is_reservation_enabled(request.site.settings),
         )
 
     @classmethod
     def clean_input(
         cls,
+        request,
```

Comment:
Usually we pass the `info` object in a case like this, if you look at other mutations. To be consistent I would change `request` to `info` in these mutations.